test/int/cluster_test.go

```diff
@@ -112,15 +112,66 @@ var _ = Describe("AtlasCluster", func() {
 		})
 	}
 
-	performUpdate := func() {
+	performUpdate := func(timeout interface{}) {
 		Expect(k8sClient.Update(context.Background(), createdCluster)).To(Succeed())
 
 		Eventually(testutil.WaitFor(k8sClient, createdCluster, status.TrueCondition(status.ReadyType), validateClusterUpdatingFunc()),
-			1200, interval).Should(BeTrue())
+			timeout, interval).Should(BeTrue())
 
 		lastGeneration++
 	}
 
+	Describe("Create cluster & change ReplicationSpecs", func() {
+		It("Should Succeed", func() {
+			expectedCluster := mdbv1.DefaultGCPCluster(namespace.Name, createdProject.Name)
+
+			By(fmt.Sprintf("Creating the Cluster %s", kube.ObjectKeyFromObject(expectedCluster)), func() {
+				createdCluster.ObjectMeta = expectedCluster.ObjectMeta
+				Expect(k8sClient.Create(context.Background(), expectedCluster)).ToNot(HaveOccurred())
+
+				Eventually(testutil.WaitFor(k8sClient, createdCluster, status.TrueCondition(status.ReadyType), validateClusterCreatingFunc()),
+					30*time.Minute, interval).Should(BeTrue())
+
+				doCommonChecks()
+				checkAtlasState()
+			})
+
+			By("Updating ReplicationSpecs", func() {
+				createdCluster.Spec.ReplicationSpecs = append(createdCluster.Spec.ReplicationSpecs, mdbv1.ReplicationSpec{
+					NumShards: int64ptr(2),
+				})
+				createdCluster.Spec.ClusterType = "SHARDED"
+				performUpdate(40 * time.Minute)
+				doCommonChecks()
+				checkAtlasState()
+			})
+		})
+	})
+
+	Describe("Create cluster & increase DiskSizeGB", func() {
+		It("Should Succeed", func() {
+			expectedCluster := mdbv1.DefaultGCPCluster(namespace.Name, createdProject.Name)
+
+			By(fmt.Sprintf("Creating the Cluster %s", kube.ObjectKeyFromObject(expectedCluster)), func() {
+				createdCluster.ObjectMeta = expectedCluster.ObjectMeta
+				Expect(k8sClient.Create(context.Background(), expectedCluster)).ToNot(HaveOccurred())
+
+				Eventually(testutil.WaitFor(k8sClient, createdCluster, status.TrueCondition(status.ReadyType), validateClusterCreatingFunc()),
+					1800, interval).Should(BeTrue())
+
+				doCommonChecks()
+				checkAtlasState()
+			})
+
+			By("Increasing DiskSizeGB", func() {
```

Comment:
[q] do you think we need the separate test for increasing DiscSize if we have the one decreasing it? https://github.com/mongodb/mongodb-atlas-kubernetes/blob/2ba4112c78a3257973e5c01dcaeb95ad8569fcf3/test/int/cluster_test.go#L189-L188

Reply:
Yes, as part of #138

Reply:
This one takes a lot of time due to disk provisioning, so I decided against including it in the "main" test suite

Reply:
as discussed p2p it may make more sense to cover something different (like `providerSettings.instanceSizeName` or whatever) as DiscSize is already covered in the other testcase)

Reply:
as discussed p2p it may make more sense to cover something different (like `providerSettings.instanceSizeName` or whatever) as DiscSize is already covered in the other testcase)